recordedit/form.controller.js

```diff
@@ -471,5 +471,108 @@
+
+        var formContainerEl = $rootElement.find('.form-container');
+
+        // Get the formedit div
+        var elem = $rootElement.find('#formEdit');
+
+        // Get height of formEdit div to use for resizing the fixed columns height
+        var elemHeight = elem.outerHeight();
+        var trs;
+
+        var scope = $rootScope;
+
+        // Set outer width of element to be less by caption column Width and add buttonWidth, 
+        // so that it doesn't scrolls due to the margin-left applied before extra padding
+        function onResize(doNotInvokeEvent) {
+            var elemWidth = formContainerEl.outerWidth();
+            vm.formEditDivMarginLeft.width = elemWidth - captionColumWidth - 30 - 15;
+            if (!doNotInvokeEvent) scope.$digest();
+        }
+        onResize(true);
+
+        // Listen to window resize event to change the width of div formEdit
+        angular.element($window).bind('resize', function() {
+            onResize();
+        });
+
+        // This function is called whenever the height of formEdit div changes
+        // This might be because of selecting/clearing something from the popup for foreighn keys
+        // It is called initially once, to adjust heights of fixed columns according to their next td element
+        function resizeColumns() {
+
+            // Set timer null to reset settimeout
+            timer = null;
+            
+            // get current height of div formEdit
+            var h = elem.height();
+
+            // If current height of div formEdit has changed than the previous one
+            if (elemHeight !== h) {
+                
+                console.log("Height changed");
+
+                // Set elemHeight to avoid changing height of first td for no change in height
+                elemHeight = h;
+                
+                // Get all rows of the table
+                if (!trs) trs = elem.find('tr.entity');
+
+                // iterate over each row
+                for(var i=0;i<trs.length;i++) { 
+                    // Get the height of the first column and  second column of the row
+                    // Which are the key and value for the row
+
+                    var keytdHeight = trs[i].children[0].offsetHeight, 
+                        valuetdHeight = trs[i].children[1].offsetHeight;  
+
+                    // If keytdHeight is greater than valuetdHeight
+                    // then set valuetdHeight
+                    // else change coltdHeight for viceversa condition
+                    if (keytdHeight > valuetdHeight) { 
+                        trs[i].children[1].height = keytdHeight;
+                    } else if (valuetdHeight > keytdHeight)  {   
+                        trs[i].children[0].height = valuetdHeight;
+                    } 
+                }
+            }
+        }
+
+        var DEBOUNCE_INTERVAL = 50; //play with this to get a balance of performance/responsiveness
```

Comment:
What is this debounce for? It looks like it's for the height because it can change based on the # of forms and the content in each form. Maybe rename the variable to be more specific. 

We are introducing a debounce into the popup modal so it will auto search after x amount of time, and don't want to confuse that debounce with this one.